fotoobo/inventory/inventory.py

```diff
@@ -174,6 +191,27 @@ def _load_inventory(self) -> None:
                 self.assets[name] = GenericDevice(**asset)
     """
 
+    def _load_data_from_vault(self, vault_dict: Dict[str, Any]) -> None:
+        """Load the credentials from a vault"""
+        log.debug("Loading credentials from vault '%s'", vault_dict["url"])
+        vault = Client(**vault_dict)
+        data = vault.get_data()
+        self.vault_data = data.get("ok", {}).get("data", {}).get("data", {})
+
+    def _replace_with_vault_data(self) -> None:
+        """Replace asset attributes with data from the vault"""
+        for name, asset in self.assets.items():
+            for attribute in dir(asset):
+                if not attribute.startswith("_"):
+                    if getattr(asset, attribute) == "VAULT":
+                        try:
+                            setattr(asset, attribute, self.vault_data[name][attribute])
+                            log.debug(
+                                "Vault attribute '%s.%s' replaced successfully", name, attribute
+                            )
+                        except KeyError:
+                            log.warning("Vault attribute '%s.%s' not found", name, attribute)
```

Comment:
Perhaps we should open up an issue, that WARNING is the default log level (if nothing set, currently default is no logging if not requested), so users will see this warning.

Reply:
https://github.com/migros/fotoobo/issues/170